sklearn/utils/_param_validation.py

```diff
@@ -315,7 +319,24 @@ def __str__(self):
         options_str = (
             f"{', '.join([self._mark_if_deprecated(o) for o in self.options])}"
         )
-        return f"a str among {{{options_str}}}"
+        return f"a {_type_name(self.type)} among {{{options_str}}}"
+
+
+class StrOptions(Options):
+    """Constraint representing a finite set of strings.
+
+    Parameters
+    ----------
+    options : set of str
+        The set of valid strings.
+
+    deprecated : set of str or None, default=None
+        A subset of the `options` to mark as deprecated in the string
+        representation of the constraint.
+    """
+
+    def __init__(self, options, deprecated=None):
```

Comment:
Do you think making `deprecated` a keyword-only argument will make this `Option` and `StrOption` more clear an intuitive?

```suggestion
    def __init__(self, options, *, deprecated=None):
```